src/common/transformations/src/transformations/utils/utils.cpp

```diff
@@ -489,14 +489,23 @@ bool is_constant_and_all_values_equal_int(const Output<Node>& output, const int6
     return false;
 }
 
-bool is_on_constant_path(const ov::Output<ov::Node>& output) {
+bool is_on_constant_path(const ov::Output<ov::Node>& output,
+                         const std::unordered_set<std::type_index>& break_node_types) {
     auto status = true;
     std::deque<ov::Node*> nodes_to_calculate = {output.get_node()};
 
     while (status && !nodes_to_calculate.empty()) {
         auto current_node = nodes_to_calculate.front();
         nodes_to_calculate.pop_front();
 
+        // Check if the current node matches any type in break_node_types
+        if (!break_node_types.empty()) {
+            std::type_index current_type(typeid(*current_node));
+            if (break_node_types.find(current_type) != break_node_types.end()) {
+                return false;
+            }
+        }
+
         if (current_node->get_input_size() == 0 && !ov::is_type<ov::op::v0::Constant>(current_node)) {
```

Comment:
It should be a bug on is_on_constant_path(). why not try more general fixing? The RandomUniform() node can't be const folded. 
```suggestion
        if ((current_node->get_input_size() == 0 && !ov::is_type<ov::op::v0::Constant>(current_node)) ||
        current_node->can_constant_fold(current_node->input_values())){
```

Reply:
Hi, @luweizhou2016 : do you mean with this changes `||
        !current_node->can_constant_fold(current_node->input_values())` ?
but it seems with this kind of changes, we will limit the condition to [constant inputs only](https://github.com/openvinotoolkit/openvino/blob/4ebe493140b90cbaaa68e2c71fe8e1dbd7461822/src/core/src/node.cpp#L711-L713), while this seems will make current 'constant_path' logic miss some cases, e.g. this specific following kind( the following second 'Eltwise' pattern should be suitable pattern for 'constant_path' logic if none the 'RandomUniform'):
![image](https://github.com/user-attachments/assets/9b5d308d-2e00-4ac5-bacb-088f525633ab)

Reply:
OK. seems can_constant_fold() can't propagate to the graph input. 
What about  " || ov::is_type(ov::op::v8::RandomUniform)", the key is this should be a bug and should be fixed in the is_on_constant_path() internally not by the API user. Once the graph path has the RandomUniform OP, it should return false.

Reply:
yes, fix in the 'is_on_constant_path()' internally can solve our issue and without change current API. But I think using current method would be easier for further extend(if some other Ops like 'RandomUniform' needed to be exclude from 'is_on_constant_path()' logic in the future, the modification will only needed in user space instead of this function). 

since synced with @luweizhou2016 offline, currently this is the only concern of this pr, @maxnick which method do you suggest?(and could you please help have a final review of this pr when you are free? thanks)

Reply:
1.  Addition about `can_constant_fold` or  `is_const_folding_disabled`
I think we expect  `return true` from  'is_on_constant_path()' if we call it on some dequantized weights
const (low precision) -> convert -> Subtract (zero_points) -> Multiply(scale) -> Convolution/Matmul etc

usually, we mark the dequantization subgraph with disable_const_folding attribute to prevent ConstantFolding transformation execution.

I believe we expect that `weights_path` will  `return true` here
<img width="426" alt="{9CAB3181-9333-40C4-B0C4-E58A89BF9458}" src="https://github.com/user-attachments/assets/75417e8c-6bcb-4fd6-ac28-5e6043444c58" />

so we can't use  `can_constant_fold`, it will return false anyway (due to disable_const_folding marking).
and we can't re-write  'is_on_constant_path()' helper using `is_const_folding_disabled`.

2. Does RandomUniform on the constant path always break Constant calculation  (const folding) logic?

Reply:
Using the direct type check may be not the best idea, as it introduces an internal coupling between the openvino operation semantics and the transformation. Checking `can_constant_fold` looks more robust, but the question is, are we really sure that all the operations returning `can_constant_fold` false should prevent MatMul->FC conversion? If so. We can run a reverse DFS from the MatMul second input (constant) up to the network input nodes (Constants or Params), which costs O(N), and then run `can_constant_fold` on all the found ops. If any of the nodes returns `can_constant_fold == false`, then we leave the MatMul untouched. What do you think?

Reply:
`can_constant_fold` returns false in case of the op inputs are not const, so we can't use it here.
we can propagate some rt_info attribute in DFS e.g. "can_be_folded" which will be depends on disable_const_folding attribute and leaf Constants 
but what about the dequantized weights case as I described above?

Reply:
Ok, that means we don't have any other working solution except checking the operation type directly.
@liubo-intel , isn't it better to use the ngraph emended RTTI and pass `ov::DiscreteTypeInfo` to check against the operation dynamic type `op->get_type_info()`?